### PR TITLE
fix: remove cra, add vite for client build

### PR DESCRIPTION
**Pull request by dpportet**
# Description

* Replace CRA with Vite for the client build to significantly reduce build time and use more modern tooling
* Should help with the cloud deployment, in my local testing it reduced cloud deployment time from >1hr to <10mins.
* Added `dynamicImport` and `commonjs` vite plugins to handle the stencil based components properly (bar chart and timeline)
   * This is specifically needed for [this code](https://github.com/ionic-team/stencil/blob/main/src/client/client-load-module.ts#L32)

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- [X] Manual tested local development and cloud deployment. Tests passing. I will do some more manual testing before merging to try to catch any weird edge-cases 

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

## Legal

This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "build": "yarn clean && turbo run build",
     "clean": "yarn install && turbo run clean && rimraf .turbo && rimraf *.tsbuildinfo",
     "clean:full": "yarn install && turbo run clean:full && rimraf .turbo && rimraf *.tsbuildinfo && rimraf node_modules",
-    "dev": "turbo run dev",
+    "dev": "yarn build && turbo run dev",
```
Comment (chejimmy) [inline package.json:19]: 1. is `yarn build` necessary?

2. it appears the `yarn build` is need for `client` only, better structure to limit this to `client` only
e.g. in `apps/client/package.json`, do `"dev": "yarn build && DISABLE_ESLINT_PLUGIN=true BROWSER=none PORT=3001 vite",`

File: apps/client/vite.config.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import dynamicImport from 'vite-plugin-dynamic-import';
+import commonjs from 'vite-plugin-commonjs';
+import path from 'path';
+
+export default defineConfig(() => {
+  return {
+    build: {
+      outDir: 'build',
+    },
+    plugins: [
```
Comment (chejimmy) [inline apps/client/vite.config.ts:12]: need to configure `app-kit` to be a singleton to utilize the metrics [current config](https://github.com/awslabs/iot-application/commit/2a8208950306351f04f550ec237a5b73e7231c7a#diff-a442e6ce9c8ed6d5ecaf9f47bff7a20278cff7d19f56c148f4a4f32bac5ea4a5)
I will help looking for the vite counterpart as well 🙏

Comment (chejimmy) [reply apps/client/vite.config.ts:12]: please try the config here https://github.com/originjs/vite-plugin-federation?tab=readme-ov-file#packagepath-string
with shared of
```
      shared: ['@iot-app-kit/core'],
```

Comment (dpportet) [reply apps/client/vite.config.ts:14]: As we discussed offline, I added 

```
  federation({
    shared: ['@iot-app-kit/core']
  }),
```

to the plugins which will create the singleton instance as expected. I tested in the cloud deployment and saw the metrics being sent.

I also had to add a plugin to remove the topLevelAwaits from core.

Comment (dpportet) [reply apps/client/vite.config.ts:12]: As we discussed offline, we're going to hold off on adding this plugin and instead find an alternative for creating a singleton app kit instance for our logging solution. With this config, we're adding top level awaits, then having to add another plugin to remove them, and it's breaking our tests. We will follow up with a more stable solution.

**Review by chejimmy — APPROVED** [review]
🚢 TY for fixing this one to make us run faster 🏎️💨
